feat(performance): Adds queues instrumentation for more platforms

edwardgou-sentry opened the pull request:
Adds custom queues instrumentation docs for java, go, ruby, and dotnet

sl0thentr0py commented:
hey @edwardgou-sentry sorry for the double work but I just added all the ruby modules here, so can you remove them from this PR?
https://github.com/getsentry/sentry-docs/pull/10245

edwardgou-sentry commented:
> hey @edwardgou-sentry sorry for the double work but I just added all the ruby modules [here](https://github.com/getsentry/sentry-docs/pull/10245), so can you remove them from this PR?

removed, thanks @sl0thentr0py !

adinauer commented on docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,104 @@
+        processMessage(message);
+    } catch (Exception e) {
+        // In case of an error set the status to "internal_error"
+        span.setStatus(SpanStatus.INTERNAL_ERROR);
```
Should we rethrow here?

adinauer replied:
addressed in https://github.com/getsentry/sentry-docs/pull/10260

adinauer commented on docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,104 @@
+    }
+}
+
+transaction.finish();
```
This should go into a `finally {}` block.

adinauer replied:
addressed in https://github.com/getsentry/sentry-docs/pull/10260

adinauer commented on docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,104 @@
+    final SentryTraceHeader traceparent = Sentry.getTraceparent();
+    final BaggageHeader baggage = Sentry.getBaggage();
+    
+    ISpan span = transaction.startChild("queue.publish", "queue_producer")
```
we should finish the span in a `finally {}` block

adinauer replied:
addressed in https://github.com/getsentry/sentry-docs/pull/10260

adinauer commented on docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,104 @@
+    
+    connection.publish(queue, message, now, traceparent, baggage);
+}
+transaction.finish();
```
we should finish the transaction in a `finally {}` block

adinauer replied:
addressed in https://github.com/getsentry/sentry-docs/pull/10260

adinauer reviewed (approved):
Thanks for taking care of this! Left some comments for the Java parts. Improving API around continuing a trace is on my list.

I've created a follow up PR to address some of my comments: https://github.com/getsentry/sentry-docs/pull/10260, feel free to merge it in if you think it makes sense.

adinauer commented on docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,104 @@
+ITransaction transaction = Sentry.startTransaction(transactionContext, "queue_consumer_transaction", "function");
+
+try {
+    ISpan span = transaction.startChild("queue.process", "queue_consumer")
```
This span is never finished, we should finish it in a `finally {}` block.

adinauer replied:
addressed in https://github.com/getsentry/sentry-docs/pull/10260

adinauer commented on docs/platforms/dotnet/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,131 @@
+
+Your `queue.process` span must exist inside a transaction in order to be recognized as a consumer span. If you are using a <PlatformLink to="/integrations/#web-frameworks">supported web framework</PlatformLink>, the transaction is created by the integration. If you use plain Python, you can start a new one using `SentrySdk.StartTransaction()`.
+
+Use `SentrySdk.ContinueTrace()` to connect your consumer spans to their associated producer spans, and `span.SetExtra()` to mark the trace of your message as success or failed.
```
When playing around we noticed that e.g. for JMS it's not possible to set `sentry-trace` as it doesn't allow the `-` character. Should we add a note like:

"Note: If you are using JMS, please transport the `sentry-trace` header with a different name like `sentry_trace` as the dash/hyphen/`-` isn't allowed in JMS."

edwardgou-sentry commented on docs/platforms/dotnet/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,131 @@
+
+Your `queue.process` span must exist inside a transaction in order to be recognized as a consumer span. If you are using a <PlatformLink to="/integrations/#web-frameworks">supported web framework</PlatformLink>, the transaction is created by the integration. If you use plain Python, you can start a new one using `SentrySdk.StartTransaction()`.
+
+Use `SentrySdk.ContinueTrace()` to connect your consumer spans to their associated producer spans, and `span.SetExtra()` to mark the trace of your message as success or failed.
```
Added a note:
![image](https://github.com/getsentry/sentry-docs/assets/83961295/da287b77-c784-41a1-9bb4-248b8942d267)

edwardgou-sentry commented on docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,104 @@
+sidebar_order: 3000
+description: "Learn how to manually instrument your code to use Sentry's Queues module. "
+---
+Sentry comes with automatic instrumentation for the most common messaging queue systems. In case yours isn't supported, you can still instrument custom spans and transactions around your queue producers and consumers to ensure that you have performance data about your messaging queues.
```
Updated! I took out the automatic instrumentation mention

edwardgou-sentry commented on docs/platforms/java/common/performance/instrumentation/custom-instrumentation/queues-module.mdx:
```diff
@@ -0,0 +1,104 @@
+| `messaging.message.retry.count ` | number | The number of times a message was attempted to be processed |
+| `messaging.message.receive.latency ` | number | The time in milliseconds that a message awaited processing in queue |
+
+Your `queue.process` span must exist inside a transaction in order to be recognized as a consumer span. If you are using a supported framework, the transaction is created by the integration. If you are using plain Java, you can start a new one using `Sentry.startTransaction()`.
```
👍 cleaned it up 
![image](https://github.com/getsentry/sentry-docs/assets/83961295/b130e7e3-3a4d-400c-ac30-15ecb60defc9)

edwardgou-sentry commented:
I'm going to merge this PR for now! We can handle any further changes to the platform docs in new PRs
